dev/tools/controllerbuilder/pkg/gocode/ast.go

```diff
@@ -73,7 +76,23 @@ type StructField struct {
 func LoadPackage(goPackage string, path string) (*Package, error) {
 	fileSet := token.NewFileSet()
 	mode := parser.ParseComments
-	var filter func(fs.FileInfo) bool
+	filter := func(info fs.FileInfo) bool {
+		p := filepath.Join(path, info.Name())
+		// Skip test files (for speed)
+		if strings.HasSuffix(info.Name(), "_test.go") {
+			return false
+		}
+		// Skip autogenerated files (in case they are broken)
+		b, err := os.ReadFile(p)
+		if err != nil {
+			klog.Warningf("failed to read file %q: %v", p, err)
```

Comment:
Question: If err, I would expect b to be empty.  If b is empty it will not 'Contains()' any string. That implies we will return true. Is that what we want?

Reply:
Switched to panic